.golangci.yml

```diff
@@ -0,0 +1,74 @@
+linters-settings:
+  govet:
+    check-shadowing: true
+  golint:
+    min-confidence: 0
+  maligned:
+    suggest-new: true
+  depguard:
+    list-type: blacklist
+    packages:
+      # logging is allowed only by logutils.Log, logrus
+      # is allowed to use only in logutils package
+      - github.com/sirupsen/logrus
+  misspell:
+    locale: US
+    ignore-words:
+      - cancelled
+  lll:
+    # Default is 120. '\t' is counted as 1 character.
+    # set our project to 200, as things like v3_structs with inline comments end up being a touch long
+    # also, for anyone using vscode, use the following configs:
+    # "rewrap.wrappingColumn": 200 ... requires the rewrap plugin
+    # "editor.rulers": [200]
+    line-length: 200
+  gocognit:
+    # minimal code complexity to report, 30 by default
+    # we need to refact the cluster resource to change this number
+    min-complexity: 94
+  nestif:
+    # minimal complexity of if statements to report, 5 by default
+    min-complexity: 7
+  goconst:
+    # minimal length of string constant, 3 by default
+    min-len: 6
+    # minimal occurrences count to trigger, 3 by default
+    min-occurrences: 6
+  gomnd:
+    settings:
+      mnd:
+        # don't include the "operation" and "assign"
+        checks: case,return
+  funlen:
+    lines: 360
+    statements: 110
+linters:
+  enable-all: true
```

Comment:
[nit] enable all is a bit risky, new version could add new linters that may fail, is safer to be explicit about enabled linters

Reply:
Maybe we can use a similar GolangCI configuration in order to have a quality`standard` between MongoDB's golang repos. Thoughts? @themantissa  @PacoDw @thetonymaster

Reply:
we totally agree, we already discuss this on slack xD